core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/stop/ClusterStopActions.java

```diff
@@ -0,0 +1,82 @@
+package com.sequenceiq.cloudbreak.core.flow2.cluster.stop;
+
+
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.statemachine.action.Action;
+
+import com.sequenceiq.cloudbreak.api.model.Status;
+import com.sequenceiq.cloudbreak.cloud.event.Selectable;
+import com.sequenceiq.cloudbreak.core.flow.context.StackStatusUpdateContext;
+import com.sequenceiq.cloudbreak.core.flow2.cluster.start.AbstractClusterStartStopAction;
+import com.sequenceiq.cloudbreak.core.flow2.cluster.start.ClusterStartStopContext;
+import com.sequenceiq.cloudbreak.core.flow2.stack.AbstractStackFailureAction;
+import com.sequenceiq.cloudbreak.core.flow2.stack.StackFailureContext;
+import com.sequenceiq.cloudbreak.domain.Stack;
+import com.sequenceiq.cloudbreak.reactor.api.event.StackEvent;
+import com.sequenceiq.cloudbreak.reactor.api.event.StackFailureEvent;
+import com.sequenceiq.cloudbreak.reactor.api.event.resource.ClusterStopRequest;
+import com.sequenceiq.cloudbreak.reactor.api.event.resource.ClusterStopResult;
+
+@Configuration
+public class ClusterStopActions {
+    private static final String STACK_STATUS = "STACK_STATUS";
+    @Inject
+    private ClusterStopService clusterStopService;
+
+    @Bean(name = "CLUSTER_STOP_STATE")
+    public Action stopCluster() {
+        return new AbstractClusterStartStopAction<StackStatusUpdateContext>(StackStatusUpdateContext.class) {
+            @Override
+            protected void doExecute(ClusterStartStopContext context, StackStatusUpdateContext payload, Map<Object, Object> variables) throws Exception {
+                Stack stack = context.getStack();
+                variables.put(STACK_STATUS, stack.getStatus());
+                clusterStopService.stopCluster(stack);
```

Comment:
clusterStopService.stopCluster() should return null if there is no cluster, and maybe AmbariClusterConnector. stopCluster() would fail later. So my suggestion is to check cluster here and send flow to failure